### PR TITLE
Updated dev docs

### PR DESCRIPTION
**Pull request by EdwinGuzman**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,3 +75,24 @@ In our AWS production environment, we have defined a CI/CD pipeline via Travis C
...
+
+## Header Component Development
+This repository deploys an instance of the Header app which is used for multiple applications as both an embedded dynamic script and as a means to distribute static HTML markup to use for scrapping by other apps.
+
+The main work for the NYPL Header is done in the component  [dgx-header-component](https://github.com/NYPL/dgx-header-component). If a change is done there and wants to be deployed to the `development` or `qa` instances of the Header app, then that change needs to be done in the `package.json` file.
+
+There are changes we'd like to deploy that can't always be tested locally for a wider audience. For example, functional and design changes to the log in menu should be deployed for the QA team on the `development` or `qa` instance. A developer can test changes in the `dgx-header-component` locally by configuring their host settings to have `localhost:3001` point to `local.nypl.org:3001` to test the log in cookie and JWT authentication. For a wider audience, we want to deploy this on `dev-header.nypl.org` for example, and then test on `dev-www.nypl.org` to be able to log in successfully.
```
Comment (ricardoom) [inline README.md]: Whats's "JWT"? Jenkins or something?

Comment (ricardoom) [inline README.md]: Also, maybe a word or two on how to configure the hosts?

Comment (EdwinGuzman) [reply README.md]: JSON web token, will update.

Comment (EdwinGuzman) [reply README.md]: Yes.

**Comment by EdwinGuzman** [comment]
Updated.

**Pull request merged**